### PR TITLE
bumped istio to 1.4.5 according to the docs to fix issue #25

### PR DESCRIPTION
**Pull request by aidun**
bumped istio to 1.4.5 according to the docs to fix issue #25

## Description
bumped istio to 1.4.5 according to the docs to fix issue #25

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

kind create cluster
arkade install istio
helm list -A
kubectl get all -n Istio-system

--> processes are running
--> helm charts in 1.4.5 are present

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by aidun** [comment]
Need x86 test k8s to test (only have rapis currently). Maybe someone can confirm that it worked.

**Comment by aidun** [comment]
Testet it with Kind. Is working as expected.

File: cmd/apps/istio_app.go [hunk 1]
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
...
 
 		if initIstio, _ := command.Flags().GetBool("init"); initIstio {
-			err = helm3Upgrade(outputPath, "istio/istio-init", namespace, "", defaultVersion, overrides, wait)
+			// Waiting for the crds to appear
+			err = helm3Upgrade(outputPath, "istio/istio-init", namespace, "", defaultVersion, overrides, true)
 			if err != nil {
 				return fmt.Errorf("unable to istio-init install chart with helm %s", err)
 			}
 
+			err = kubectl("-n", "istio-system", "wait", "--for=condition=complete", "job", "--all")
```
Comment (alexellis) [inline cmd/apps/istio_app.go]: Why is this required? Isn't it part of the `--wait` flag that is optional on apps like `openfaas` and `inlets-operator`?

File: cmd/apps/istio_app.go [hunk 2]
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
...
+
+			if err != nil {
+				return fmt.Errorf("error waiting for the crds: %s", err)
```
Comment (alexellis) [inline cmd/apps/istio_app.go]: Should be: `CRDs`

File: cmd/apps/istio_app.go [hunk 3]
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
...
+			}
+		}
+
+		if err != nil {
```
Comment (alexellis) [inline cmd/apps/istio_app.go]: Why do we have the same error twice, is this a different step?

File: cmd/apps/istio_app.go [hunk 4]
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
 		outputPath := path.Join(chartPath, "istio")
 
 		if initIstio, _ := command.Flags().GetBool("init"); initIstio {
-			err = helm3Upgrade(outputPath, "istio/istio-init", namespace, "", defaultVersion, overrides, wait)
+			// Waiting for the crds to appear
+			err = helm3Upgrade(outputPath, "istio/istio-init", namespace, "", defaultVersion, overrides, true)
```
Comment (alexellis) [inline cmd/apps/istio_app.go]: Please use the wait flag instead of forcing wait to be true.

I'm OK if you want to add a flag and set it to default a true value. See the other apps for examples or ask in #arkade on Slack if you need help with that.

File: cmd/apps/istio_app.go (hunk 1 above)
Comment (aidun) [reply cmd/apps/istio_app.go]: This are two different points.

I will remove the `kubectl` command, it is not needed, but we have to force wait here, if not the second install step will fail. Is it okay for you?

File: cmd/apps/istio_app.go (hunk 2 above)
Comment (aidun) [reply cmd/apps/istio_app.go]: Done

File: cmd/apps/istio_app.go (hunk 4 above)
Comment (aidun) [reply cmd/apps/istio_app.go]: If I don´t force and to wait, the second step will always fail. I think its a bad behavior.

File: cmd/apps/istio_app.go (hunk 3 above)
Comment (aidun) [reply cmd/apps/istio_app.go]: i removed it. copy paste error.

**Comment by aidun** [comment]
Thank you for the feedback. I followed your comments and fixed the most of it. I have only concerns to implement the optional wait on the first step. I added my explanation to your inline review.
